include/openssl/ec.h

```diff
@@ -392,15 +392,31 @@ OPENSSL_EXPORT int EC_GROUP_set_generator(EC_GROUP *group,
                                           const BIGNUM *cofactor);
 
 
-// EC_POINT_point2bn converts an |EC_POINT| to a |BIGNUM| by serializing the
-// point into the X9.62 form given by |form| then interpretting it as a BIGNUM.
+// EC_POINT_point2bn serialises |point| into the X9.62 form given by |form|
+// and returns the |BIGNUM| representation of the serialised output.
 // On success, it returns the BIGNUM pointer supplied or, if |ret| is NULL,
 // allocates and returns a fresh |BIGNUM|. On error, it returns NULL. The |ctx|
 // argument may be used if not NULL.
+//
+// Note: |EC_POINT| serialization formats are not individual big-endian
+// integers, so these aren't particularly useful. Use |EC_POINT_point2oct|
+// instead.
 OPENSSL_EXPORT OPENSSL_DEPRECATED BIGNUM *EC_POINT_point2bn(
     const EC_GROUP *group, const EC_POINT *point, point_conversion_form_t form,
     BIGNUM *ret, BN_CTX *ctx);
 
+// EC_POINT_bn2point is like |EC_POINT_point2bn|, but takes the |BIGNUM|
+// representation and de-serialises it back to an |EC_POINT|. On success,
+// it returns the EC_POINT pointer supplied or, if |ret| is NULL, allocates and
+// returns a fresh |EC_POINT|. On error, it returns NULL. The |ctx| argument
+// may be used if not NULL.
+//
+// Note: |EC_POINT| serialization formats are not individual big-endian
```

Comment:
I still don't understand how is this supposed to work at all :)
A point is `(x, y)` with `x` and `y` big integers. So how do you get from `|bn|` to `|point|`?

Reply:
oh boy, I get it now. terrible API, but what to do... I like the updated documentation.

Reply:
Yeah everything about the function really throws people off unless you look into what it's actually doing... Thanks for iterating on this, I think it's in a much better state now.